boot/stage2.c

```diff
@@ -8,6 +8,9 @@
 #include <serial.h>
 #include <drivers/ata.h>
 
+//#define STAGE2_DEBUG
```

Comment:
At a minimum, stage2 shouldn't be ingesting all the root fs meta just to load the kernel. I'm in favor of keeping the kernel in a separate boot filesystem, or even just pasting the kernel into the image as a linear region after the boot image.

Reply:
i think a straightforward concatenation would have alot of packaging benefits. you could could have all your test and application images just latent and pickled, and apply them against any arbitrary kernels. and..the applications are exactly tfs images. and since we already have userspace tools to manipulate them, that would let you do the docker 'layer' things, or similar messing about